### PR TITLE
dev-python/mongomock: MongoDB testing library (new package)

### PR DESCRIPTION
**Pull request by vpayno**
@zmedico @gyakovlev @chutz

### CODE REVIEW

**Comment by vpayno** [comment]
Needs: https://github.com/gentoo/gentoo/pull/15220

File: dev-python/mongomock/mongomock-3.19.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+RDEPEND="
+	>=dev-python/pymongo-3.10.1[${PYTHON_USEDEP}]
+	>=dev-python/sentinels-1.0.0[${PYTHON_USEDEP}]
+	>=dev-python/six-1.14.0[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		${RDEPEND}
+		>=dev-python/tox-3.14.4[${PYTHON_USEDEP}]
```
Comment (Alessandro-Barbieri) [inline dev-python/mongomock/mongomock-3.19.0.ebuild:30]: undesired dependency

File: dev-python/mongomock/mongomock-3.19.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+DESCRIPTION="Library to help testing Python code that interacts with MongoDB via Pymongo."
+HOMEPAGE=" https://github.com/mongomock/mongomock https://pypi.org/project/mongomock/"
+SRC_URI="https://github.com/mongomock/mongomock/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test"
+RESTRICT="test" # it has a lot of network tests
```
Comment (Alessandro-Barbieri) [inline dev-python/mongomock/mongomock-3.19.0.ebuild:20]: you can selectively disable the failing tests using `pytest --deselect path/file.py::test`

File: dev-python/mongomock/mongomock-3.19.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	>=dev-python/sentinels-1.0.0[${PYTHON_USEDEP}]
+	>=dev-python/six-1.14.0[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		${RDEPEND}
+		>=dev-python/tox-3.14.4[${PYTHON_USEDEP}]
+	)"
+
+FEATURES+=" network-sandbox"
```
Comment (gyakovlev) [inline dev-python/mongomock/mongomock-3.19.0.ebuild:33]: leftover from testing?

File: dev-python/mongomock/mongomock-3.19.0.ebuild (hunk 1 above)
Comment (gyakovlev) [inline dev-python/mongomock/mongomock-3.19.0.ebuild:30]: yeah need to patch out tox.
it's used by upstreams to run under multiple implementations, eclasses do it for us.

**Comment by vpayno** [comment]
Discovered that the mongo/network tests can be disabled by setting ``.

Tests are now passing.
```
============ 422 passed, 259 skipped, 309 warnings in 1.61 seconds =============
 * python3_6: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python3_7: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python3_8: running distutils-r1_run_phase _distutils-r1_clean_egg_info
>>> Completed testing dev-python/mongomock-3.19.0
```

Had to remove one assert/test case because it doesn't work with Python 2.7 and Python 3.{6..8}.

File: dev-python/mongomock/mongomock-3.19.0.ebuild (hunk 3 above)
Comment (vpayno) [reply dev-python/mongomock/mongomock-3.19.0.ebuild:30]: Now that tests have been re-enabled, should I leave this in or remove it?

File: dev-python/mongomock/mongomock-3.19.0.ebuild (hunk 2 above)
Comment (vpayno) [reply dev-python/mongomock/mongomock-3.19.0.ebuild]: The tests are mixed in. Found out I could set `NO_LOCAL_MONGO=1` to disable them.

File: dev-python/mongomock/mongomock-3.19.0.ebuild (hunk 1 above)
Comment (vpayno) [reply dev-python/mongomock/mongomock-3.19.0.ebuild]: yeah, missed that when I switched from tox to pytest.

**Pull request closed without merging**